src/pages/PictureUploadBox.tsx

```diff
@@ -0,0 +1,145 @@
+import { useTranslation } from 'react-i18next';
+
+import { Button, ButtonVariants } from '@/ui/Button';
+import { Icons } from '@/ui/Icon';
+import { Image } from '@/ui/Image';
+import { Inline } from '@/ui/Inline';
+import type { StackProps } from '@/ui/Stack';
+import { getStackProps, Stack } from '@/ui/Stack';
+import { Text, TextVariants } from '@/ui/Text';
+import { getValueFromTheme } from '@/ui/theme';
+import { Title } from '@/ui/Title';
+
+const getValue = getValueFromTheme('pictureUploadBox');
+
+type ImageType = {
+  parsedId: string;
+  description: string;
+  copyrightHolder: string;
+  thumbnailUrl: string;
+  isMain: boolean;
+  file: File;
+};
+
+type Props = StackProps & {
+  images: ImageType[];
+  onClickEditImage: (id: string) => void;
+  onClickDeleteImage: (id: string) => void;
+  onClickSetMainImage: (id: string) => void;
+  onClickAddImage: () => void;
+};
+
+const PictureUploadBox = ({
+  images,
+  onClickEditImage,
+  onClickDeleteImage,
+  onClickSetMainImage,
+  onClickAddImage,
+  ...props
+}: Props) => {
+  const { t } = useTranslation();
+
+  return (
+    <Stack spacing={2}>
+      <Title size={3}>{t('pictures.title')}</Title>
+      <Stack
+        flex={1}
+        spacing={4}
+        padding={4}
+        backgroundColor={getValue('backgroundColor')}
+        justifyContent="center"
+        css={`
+          border: 1px solid ${getValue('borderColor')};
+        `}
+        {...getStackProps(props)}
+      >
+        <Stack
+          spacing={2}
+          maxHeight={380}
+          css={`
+            overflow: auto;
+          `}
+        >
+          {images.map((image, index, imagesArr) => {
+            const thumbnailSize = 80;
```

Comment:
I think you can define this `const`outside the `.map`,  since it's the same for all the thumbnails.

Reply:
Yep nice catch